Fix configuration documentation for email duration limiter (#3365)
treeverse/lakeFS: docs/reference/configuration.md

+ `email.password` `(string)` - A string representing the password of the account. It's recommended to provide this value at runtime from a secret vault of some sort.
+ `email.local_name` `(string)` - A string representing the hostname sent to the SMTP server with the HELO command. By default, "localhost" is sent.
+ `email.sender` `(string)` - A string representing the email account which is set as the sender.

> + `email.limit_every_duration` `(duration : 1m)` - Average time to wait between sending emails. The zero value means *infinite* duration and therefore *unlimited* emails can be sent.

Reviewer: ```suggestion
+ `email.limit_every_duration` `(duration : 1m)` - The average time between sending emails. If zero is entered, there is no limit to the amount of emails that can be sent.
```

Author: Done

> + `email.burst` `(int: 10)` - Maximal burst of emails before applying `limit_every`. The zero value means no burst and therefore no emails can be sent.

Reviewer: What is limit_every?

Author: It's referring to `limit_every_duration`  (in the previous line)
